space-wizards/space-station-14
# Stationary teleporters

## About the PR
A new Gateways option has been added as an alternative system. 
![image](https://github.com/user-attachments/assets/92cd92eb-8552-498f-859e-df267ce11a6f)
<img width="706" height="340" alt="image" src="https://github.com/user-attachments/assets/4b48dd86-45ca-43b3-80de-c533957a149e" />

This system is implied to be player-controlled. It comes along with several new elements:
- Gateways chips: Data carriers of gateway coordinates. Disposable: they can be used to record coordinates, but cannot be erased. In the future could be used as loot for Salvage, with planet coordinates. 
- Gateways control console: A console into which you can insert chips with Gateways coordinates. Once the chips are inserted, the console can be used to link and control the routes of the Gateways in any way. Locked to Command Access.
- Autolinking for chips and gateways. Chips and Gateways with the same key are automatically linked when the map is initialized, which makes it convenient to use them for mapping

At the moment, neither the console nor the chips can be obtained by players, but after testing, it could be T3 technology.
Such a gateway system is emergent enough that mechanics could be added to it.
Possible potential scenarios:
- Traitors steal chips to destabilize the portals.
- Nuclear operatives secretly read the coordinates of the gateways, and directly break into the center of the station through them. (Gate-ops?)

## Why / Balance
Primarily as the core mechanics for the Gate map #32032
With the option to develop a Salvage Proposal in future 


## Basic Testing plan

- Create 2 `StationTeleporter`, `ComputerStationTeleportersControl`, and 2 `TeleporterChipBlank` (basic entities created by players)
- Click on different teleporters with the chips to link their coordinates.
- Place the chips with linked coordinates inside the console.
- Open the console (commander access required) and verify that both portals are now visible inside the console and can be linked to each other.
- Optionally, you can place the chips in `ComputerSyndicateTeleportersControl` and open the portals to ensure they are red.
- You can also verify that turning off the power shuts down the portals, and that they are restored once the power is turned back on.

### Interacting with HandTeleporter

- Create 2 portals using `HandTeleporter`
- Place the teleporter inside `ComputerStationTeleportersControl`
- Verify that the console displays the portals’ locations, that they are linked, and that they can be relinked to other portals.

### Testing Auto Link for mapping.

- Create any number of teleporters from different sections, such as `StationTeleporterSalvage`
- Create a `StationTeleporter` that does not have auto-linking enabled.
- Create a `ComputerStationTeleportersControlFilled` console.
- Verify that the console is populated with chips and linked to the teleporters created earlier, except for the teleporter without auto-linking.

## Media
![image](https://github.com/user-attachments/assets/ca3c9e53-ee84-4d5a-9be9-33a3da119433)

https://github.com/user-attachments/assets/b346aa72-e05a-4225-aedf-57f3c5239591

## Requirements



:cl:
- add: New T3 “TeleSci” technology added. Allows you to create a portal control console, a stationary teleporter, and coordinate chips.
